lib/sqlalchemy/sql/schema.py

```diff
@@ -2758,9 +2758,11 @@ class ForeignKey(DialectKWArgs, SchemaItem):
 
     _table_column: Optional[Column[Any]]
 
+    _colspec: Union[str, Column[Any], Callable[[], _DDLColumnArgument]]
+
     def __init__(
         self,
-        column: _DDLColumnArgument,
+        column: Union[_DDLColumnArgument, Callable[[], _DDLColumnArgument]],
```

Comment:
**mike bayer** (zzzeek) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4831